tools/clang/tools/dxildll/dxcvalidatorp.cpp

```diff
@@ -0,0 +1,287 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// dxcvalidatorp.cpp                                                         //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Implements the DirectX Validator object, including signing support.       //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "llvm/Bitcode/ReaderWriter.h"
+#include "llvm/IR/DiagnosticPrinter.h"
+#include "llvm/IR/LLVMContext.h"
+
+#include "dxc/DxilContainer/DxilContainer.h"
+#include "dxc/HLSL/DxilValidation.h"
+#include "dxc/Support/WinIncludes.h"
+#include "dxc/Support/microcom.h"
+
+#include "dxc/DxilRootSignature/DxilRootSignature.h"
+#include "dxc/Support/FileIOHelper.h"
+#include "dxc/Support/Global.h"
+#include "dxc/Support/dxcapi.impl.h"
+#include "dxc/Support/microcom.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/MSFileSystem.h"
+#ifdef _WIN32
+#include "dxc/Tracing/dxcetw.h"
+#endif
+#include "DxilHash.h"
+
+using namespace llvm;
+using namespace hlsl;
+
+class DxcValidator : public IDxcValidator2, public IDxcVersionInfo {
+private:
+  DXC_MICROCOM_TM_REF_FIELDS()
+
+  HRESULT RunValidation(_In_ IDxcBlob *pShader,
```

Comment:
Can we get rid of the SAL annotations? We did from everywhere in DXC except the public API headers.

Reply:
Fixed.